cmd/tsdb/main.go

```diff
@@ -383,3 +396,89 @@ func getFormatedTime(timestamp int64, humanReadable *bool) string {
 	}
 	return strconv.FormatInt(timestamp, 10)
 }
+
+func analyzeBlock(b *tsdb.Block) {
+	fmt.Printf("Block: %s\n", b.Dir())
+	meta := b.Meta()
+	fmt.Printf("Series: %d\n", meta.Stats.NumSeries)
+	ir, err := b.Index()
+	if err != nil {
+		exitWithError(err)
+	}
+	defer ir.Close()
+
+	ln, err := ir.LabelNames()
+	if err != nil {
+		exitWithError(err)
+	}
+	fmt.Printf("LabelNames: %d\n", len(ln))
+
+	type postingInfo struct {
+		key    string
+		metric uint64
+	}
+	postingInfos := []postingInfo{}
+
+	printInfo := func() {
+		sort.Slice(postingInfos, func(i, j int) bool { return postingInfos[i].metric > postingInfos[j].metric })
+
+		for i, pc := range postingInfos {
+			fmt.Printf("%d %s\n", pc.metric, pc.key)
+			if i >= 20 {
+				break
+			}
+		}
+	}
+
+	for _, n := range ln {
+		lv, err := ir.LabelValues(n)
+		if err != nil {
+			exitWithError(err)
+		}
+		postingInfos = append(postingInfos, postingInfo{n, uint64(lv.Len())})
+	}
+	fmt.Printf("\nHighest Cardinality Labels:\n")
+	printInfo()
+
+	labelpairsUncovered := map[string]uint64{}
+	labelpairsCount := map[string]uint64{}
+	entries := 0
+	p, err := ir.Postings("", "") // The special all key.
+	if err != nil {
+		exitWithError(err)
+	}
+	lbls := labels.Labels{}
+	chks := []chunks.Meta{}
+	for p.Next() {
+		err = ir.Series(p.At(), &lbls, &chks)
+		// Amount of the block time range not covered by this series.
+		uncovered := uint64(meta.MaxTime-meta.MinTime) - uint64(chks[len(chks)-1].MaxTime-chks[0].MinTime)
+		for _, lbl := range lbls {
+			key := lbl.Name + "=" + lbl.Value
+			labelpairsUncovered[key] += uncovered
+			labelpairsCount[key] += 1
+			entries += 1
+		}
+	}
+	if p.Err() != nil {
+		exitWithError(p.Err())
+	}
+	fmt.Printf("Postings (unique label pairs): %d\n", len(labelpairsUncovered))
+	fmt.Printf("Postings entries: %d\n", entries)
+
+	postingInfos = postingInfos[:0]
+	for k, m := range labelpairsCount {
+		postingInfos = append(postingInfos, postingInfo{k, m})
+	}
+
+	fmt.Printf("\nMost Common Label Pairs:\n")
+	printInfo()
```

Comment:
probably best to have this as `printInfo(postingInfos)` to be a bit more explicit.